[16.0][FIX] partner_auto_archive: Add mail dependency to fix tracking parameter warning

mb-ife opened the pull request:
When installing or upgrading the module, a warning appeared about the unknown 
'tracking' parameter. This fix adds mail as a dependency to properly support 
the tracking parameter on fields, ensuring clean installation and upgrades 
without warnings.

ng-ife reviewed (approved):
Finally found the issue. It's dependencies.
--edit--
It solves the issue. Still I would like to know why this problem occurs as res.partner has a chatter and should therefor be able to track.

ng-ife commented:
Does not occur anymore. Probably related to another module or fixed in odoo core.

Commit:
[16.0][FIX] partner_auto_archive: Add mail dependency to fix tracking parameter warning

When installing or upgrading the module, a warning appeared about the unknown
'tracking' parameter. This fix adds mail as a dependency to properly support
the tracking parameter on fields, ensuring clean installation and upgrades
without warnings.

ng-ife reviewed (approved):
LGTM

ng-ife commented:
Hello @dreispt 
I saw you already merged changes for this module before. Can you have a look here?
Thank you

dreispt commented on partner_stage/__manifest__.py:
```diff
@@ -9,7 +9,7 @@
     "version": "16.0.1.0.1",
     "license": "AGPL-3",
-    "depends": ["contacts"],
+    "depends": ["contacts", "mail"],
```
Doesn't contacts already depend on mail?
Otherwise it wouldn't be capable to provide the Chatter feature.

ng-ife replied:
@dreispt You're right. Still we have the issue that tests (e.x. on Odoo.sh fail) if we don't add this here. What about changing the dependency to mail instead of contacts. Contacts is not technically needed for the module to work.
